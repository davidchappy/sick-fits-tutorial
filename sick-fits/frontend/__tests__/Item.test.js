import ItemComponent from '../components/Item'
import { shallow } from 'enzyme'

const fakeItem = {
  id: 'ABC123',
  title: 'A cool item',
  description: 'This item',
  price: 5000,
  image: 'dog.jpg',
  largeImage: 'largedog.jpg'
}

describe('<Item />', () => {
  const wrapper = shallow(<ItemComponent item={fakeItem}/>)

  it ('renders the image properly', () => {
    const img = wrapper.find('img')
    expect(img.props().src).toBe(fakeItem.image)
    expect(img.props().alt).toBe(fakeItem.title)
  })

  it('renders the price tag and title', () => {
    const PriceTag = wrapper.find('PriceTag')
    expect(PriceTag.children().text()).toBe('$50')

    expect(wrapper.find('Title a').text()).toBe(fakeItem.title)
  })

  it('renders out the buttons separately', () => {
    const buttonList = wrapper.find('.buttonList')
    console.log(buttonList.debug())

    expect(buttonList.children()).toHaveLength(3)
    expect(buttonList.find('Link')).toHaveLength(1)
    expect(buttonList.find('AddToCart').exists()).toBe(true)
    // expect(buttonList.find('DeleteItem')).toHaveLength(1)
  })
})