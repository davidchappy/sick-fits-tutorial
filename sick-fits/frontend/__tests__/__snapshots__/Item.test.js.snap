// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A cool item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        A cool item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $40
  </PriceTag>
  <p>
    This item
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="ABC123"
    />
    <componentName
      id="ABC123"
    >
      Delete this Item
    </componentName>
  </div>
</ItemStyles__Item>
`;
